kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/spec/lifecycle.kt

```diff
@@ -89,12 +91,27 @@ suspend fun TestCase.invokeAllAfterTestCallbacks(result: TestResult): Try<TestCa
       Try.Failure(it)
    }, { listeners ->
       Try {
+         var currentResult = result
+         var currentException: Error? = null
          listeners.forEach {
-            it.afterTest(this, result)
-            it.afterAny(this, result)
-            if (type == io.kotest.core.test.TestType.Test) it.afterEach(this, result)
-            if (type == io.kotest.core.test.TestType.Container) it.afterContainer(this, result)
+            try {
+               it.afterTest(this, currentResult)
+               it.afterAny(this, currentResult)
+               if (type == TestType.Test) it.afterEach(this, currentResult)
+               if (type == TestType.Container) it.afterContainer(this, currentResult)
+            } catch (e: java.lang.Error) {
+               if (!listOf(TestStatus.Failure, TestStatus.Error).contains(currentResult.status)) {
```

Comment:
if we also have the error in test -> we don't make anything with this because the test already red (we just need to give other listeners try to finished their 'afterTest')